extensions/ql-vscode/src/query-server/new-messages.ts

```diff
@@ -29,6 +29,35 @@ export interface ClearCacheParams {
    * Whether the cache should actually be cleared.
    */
   dryRun: boolean;
+  /**
+   * The mode to use when trimming the disk cache.
+   */
+  mode?: CacheTrimmingMode;
+}
+
+export type CacheTrimmingMode = number;
+/**
+ * The mode to use when trimming the disk cache. This namespace is intentionally not an enum, see
+ * "for the sake of extensibility" comment above.
+ */
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace CacheTrimmingMode {
+  /** The entire cache is deleted unconditionally. */
+  export const BRUTAL = 0;
+  /** Only `cached` predicates are kept. */
+  export const NORMAL = 1;
+  /**
+   * Trim the cache down to the configured size, but *within* this limit keep anything that
+   * appears to be even possibly potentially useful in the future.
+   */
+  export const LIGHT = 2;
```

Comment:
Are there plans to support `LIGHT` and `GENTLE` in the future?

Reply:
Probably not, as they're pretty obscure and there's little practical reason to support them in VSCode. Also, we're renaming "brutal" → "clear" and "normal" → "trim" to be more in line with the VSCode commands, so if you have suggestions on new names for "light" and "gentle", I'm all ears :)